service/src/main/java/se/inera/statistics/service/processlog/EnhetManager.java

```diff
@@ -62,17 +62,18 @@
 
     @Transactional
     public int updateName(ListGetHsaUnitsResponseType.HsaUnits hsaUnits) {
-        return hsaUnits.getHsaUnit().stream().reduce(0, (integer, hsaUnit) -> {
-            final Query q = manager.createNativeQuery("UPDATE enhet SET namn = :namn WHERE enhetId = :id AND namn <> :namn");
+        return hsaUnits.getHsaUnit().stream().reduce(0, (partialUpdateResultCount, hsaUnit) -> {
+            final Query query = manager.createQuery(
+                    "UPDATE Enhet e SET e.namn = :namn WHERE e.enhetId = :id AND e.namn <> :namn");
```

Comment:
Den här är jag inte riktigt kompis med. Reflekterade över denna när jag först felsökte problemet som skapade denna Jira. Känns mindre optimalt att ha en SQL sats för sig själv i en service. Detta bör nog hanteras i persistenslagret. Dvs ta in en referens som man använder.

Hur känner du inför denna tanke.

Reply:
Den låg tidigare i UpdateEnhetNamnFromHsaFileService. Jag har flyttat den till EnhetManager. Statistik verkar inte ha något persistence-lager, iaf inte i paketstruktur. Entityn Enhet ligger i samma paket som EnhetManager.

Reply:
Aha. Men då känns det ju lugnt. Såg inte att du hade flyttat den 👍